### ISSUE TITLE
Line, which starts from "elif" or "else" moves one indent level left

### ISSUE THREAD

**Issue by vlad1777d** [atom/language-python#209]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When line starts from "elif" or "else" keyword, has some indentation (I use tabs), before it already used "return" statement and you add something to it, then it automatically moves to one indent level left.

### Steps to Reproduce

0. Maybe to check preferences to be consistent with myne. Also to use tab key (and tabs) for indentation.
1. Make some block with indentation
2. Add line with "if" on it; and it's block
2.5. Include "return" statement into "if" block (or "elif" block, which is previous to that, which will be unindented).
3. Add line with "elif" or "else" and something into it. 
4. After you'll press ":", than this line will be moved to one indentation level left.
5. After this if you'll move this line to one indentation right - it'll be moved
6. If you'll try to change something on this line - it'll be moved to one indentation level left again.

**Expected behavior:** Nothing would be moved to one indentation level left. Because functions often have several "return" statements.
**Actual behavior:** Line is moved to one indentation level left

**Reproduces how often:** Always (or often)

### Versions

Atom    : 1.20.0
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0

apm  1.18.4
npm  3.10.10
node 6.9.5 x64
python 2.7.14rc1
git 2.14.1

OS: Debian Sid x64
Linux: 4.12.0-1-amd64

**Atom's preferences**:
____Editor tab:
________Auto Indent: True
________Auto Indent On Paste: True
________Soft Tabs: False
________Tab Type: Auto

### Additional Information

**Video** with reproduction of this bug on YouTube: https://youtu.be/I-b7O0y13Js
In 1.20.1 this error is present too.

**Comment by lumbduck** [comment]
I've had this problem as well. The unwanted shifting of the indent happens in several situations while editing `else` and `elif` statements.

I made sure that all issue-tracking prerequisites were met. One simple example follows.

### Steps to Reproduce

1. Launch `atom --safe` with default configuration (I moved my configuration to a backup)
2. Unzip and open the following python file in the editor: 
[Atom_indent_issue.py.zip](https://github.com/atom/language-python/files/1342077/Atom_indent_issue.py.zip)
3. Place cursor on Line 4, anywhere after the 9th character (inside the condition of the line beginning with `elif`)
4. Begin editing the conditional statement

**Expected behavior:** I would expect the line to stay stationary as I add new content in the middle of it.

**Actual behavior:** The indentation of line 4 shifts to the left

**Reproduces how often:** Always

### Versions
On Mac OSX (10.12.6)
Atom    : 1.20.1
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0

### Additional Information
You can find many similar variations of this issue. For example, in step 3-4 above, instead simply highly `elif` and type `else` in its place. The indentation will shift the same way.

**Comment by csvoss** [comment]
This issue is happening to me as well. It seems to happen when my `elif` or `else` is directly after a `return` or `pass` in the previous block.

**Comment by oceanicdev** [comment]
I have same issue in next cases.
```
def fname(arg):
    if True:
        pass
else:
```

```
def fname(arg):
    pass
<- If click to enter after 'pass' then a new line will be here
```

**Comment by Hanzhizheng** [comment]
same issue

**Comment by eicksl** [comment]
I've also been experiencing this issue, both before and after the new 1.21.1 update.

**Comment by Jglez2330** [comment]
same problem

**Comment by Navigat-r** [comment]
I'm having this exact same problem. Upon finishing the line of an elif or else statement with a colon, that line unindents itself. This seems to have been an issue for quite some time, how come it's still a problem? Is it a mysterious or difficult problem to solve?

**Comment by csvoss** [comment]
Thanks to 50Wliu's pull request https://github.com/atom/language-python/pull/212, this is now fixed in language-python [release](https://github.com/atom/language-python/releases) 0.45.5, which appears to be included in core Atom [release](https://github.com/atom/atom/releases) 1.23.0-beta1. If you want to get this fix, you can manually install 1.23.0-beta1 and it will be fixed! (I have tested this; it was fixed in 1.23.0-beta1 but not in 1.22.0.)

**Comment by Navigat-r** [comment]
I tried it out, updating to the beta, and it seems indeed like it works! Thanks for the tip! :)
